CMakeLists.txt

```diff
@@ -31,29 +31,38 @@ if (MATERIALX_BUILD_JS)
     endif()
 endif()
 
+option(MATERIALX_BUILD_IOS "Build the MaterialX for iOS." OFF)
+if (MATERIALX_BUILD_IOS)
+    set(CMAKE_SYSTEM_NAME iOS)
+    add_definitions(-DTARGET_OS_IOS=1)
+endif()
+
 project(MaterialX)
 
+if(NOT MATERIALX_BUILD_IOS)
 option(MATERIALX_BUILD_PYTHON "Build the MaterialX Python package from C++ bindings. Requires Python 3.6 or greater." OFF)
 option(MATERIALX_BUILD_VIEWER "Build the MaterialX Viewer." OFF)
 option(MATERIALX_BUILD_GRAPH_EDITOR "Build the MaterialX Graph Editor." OFF)
 option(MATERIALX_BUILD_DOCS "Create HTML documentation using Doxygen. Requires that Doxygen be installed." OFF)
-
 option(MATERIALX_BUILD_GEN_GLSL "Build the GLSL shader generator back-end." ON)
 option(MATERIALX_BUILD_GEN_OSL "Build the OSL shader generator back-end." ON)
 option(MATERIALX_BUILD_GEN_MDL "Build the MDL shader generator back-end." ON)
```

Comment:
To maintain the clarity of this top-level CMakeLists.txt file, it might be better to leave this list of build options unmodified, and to simply force individual values to false if they're not currently supported on iOS.

One additional thought is that these language generators (GLSL/OSL/MDL) ought to be supportable on iOS, since they don't require any language-specific libraries or rendering facilities, so you might consider leaving them available for users to enable on iOS.